lib/ansible/cli/galaxy.py

```diff
@@ -969,39 +991,115 @@ def _display_role(gr):
+        if context.CLIARGS['type'] == 'role':
+            if context.CLIARGS['role']:
+                # show the requested role, if it exists
+                name = context.CLIARGS['role']
+                gr = GalaxyRole(self.galaxy, self.api, name)
+                if gr.metadata:
+                    display.display('# %s' % os.path.dirname(gr.path))
+                    _display_role(gr)
+                else:
+                    display.display("- the role %s was not found" % name)
             else:
-                display.display("- the role %s was not found" % name)
-        else:
-            # show all valid roles in the roles_path directory
-            roles_path = context.CLIARGS['roles_path']
-            path_found = False
-            warnings = []
-            for path in roles_path:
-                role_path = os.path.expanduser(path)
-                if not os.path.exists(role_path):
-                    warnings.append("- the configured path %s does not exist." % role_path)
-                    continue
-                elif not os.path.isdir(role_path):
-                    warnings.append("- the configured path %s, exists, but it is not a directory." % role_path)
-                    continue
-                display.display('# %s' % role_path)
-                path_files = os.listdir(role_path)
+                # show all valid roles in the roles_path directory
+                roles_path = context.CLIARGS['roles_path']
+                for path in roles_path:
+                    role_path = os.path.expanduser(path)
+                    if not os.path.exists(role_path):
+                        warnings.append("- the configured path %s does not exist." % role_path)
+                        continue
+                    elif not os.path.isdir(role_path):
+                        warnings.append("- the configured path %s, exists, but it is not a directory." % role_path)
+                        continue
+                    display.display('# %s' % role_path)
+                    path_files = os.listdir(role_path)
+                    path_found = True
+                    for path_file in path_files:
+                        gr = GalaxyRole(self.galaxy, self.api, path_file, path=path)
+                        if gr.metadata:
+                            _display_role(gr)
+
+        elif context.CLIARGS['type'] == 'collection':
+            collections_search_paths = context.CLIARGS['collections_path']
+            collection_name = context.CLIARGS['collection']
+
+            # TODO: Should we support listing all collections in a given namespace?
+
+            # list a specific collection
+            if collection_name:
+                for path in collections_search_paths:
+                    collection_path = GalaxyCLI._resolve_path(path)
+                    if not os.path.exists(path):
+                        if path in C.COLLECTIONS_PATHS:
+                            # don't warn for missing default dirs
```

Comment:
We don't do this for `roles`, meaning we show warnings when the default directories do not exist. I found that noisy, but maybe we should warn to be consistent? Or remove the warning when checking `roles`. I would like them at least be consistent.

Reply:
I'm totally open to massaging the output format, but I agree they should be consistent.